Bug/7918 fix twistd daemon block

kalikaneko opened the pull request.

meskio commented on src/leap/common/events/examples/README.txt:
```diff
@@ -0,0 +1,49 @@
+T 127.0.0.1:33122 -> 127.0.0.1:9000 [AP]
+.+.MESSAGE........o...*M..,....
+.r..w..[.GwcU
+###
```
Will be great if we could convert this into a test to put in our CI infrastructure.

kalikaneko replied:
I have added a small tests that just checks that the authenticator has been called.

meskio commented:
Most of the tests fail in my machine:
https://share.riseup.net/#bz5_LjDp9fW5kf6JH4oWFg

meskio commented on src/leap/common/events/auth.py:
```diff
@@ -0,0 +1,92 @@
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+ZAP authentication, twisted style.
```
Should this somehow belong to txzmq? I know is not in a really active development, but maybe we could try to pull-req it to them?

kalikaneko replied:
we might send it, but it will need more work to properly implement the whole ZAP protocol. here it's only implemented the bits that we use.

meskio commented on src/leap/common/events/server.py:
```diff
@@ -14,31 +14,27 @@
 from leap.common.events.zmq_components import TxZmqServerComponent
 
+import txzmq
+
```
Is there a reason for the import move? I kind of like it more where it was.

meskio replied:
I guess there is a relation with this `TxZmqServerComponent`, but I don't see it really explicit. Maybe a comment explaining this will help.

kalikaneko replied:
no reason, actually the previous order was better. might be that I removed and had to put it in place again.

kalikaneko replied:
fixed

meskio commented on src/leap/common/events/txclient.py:
```diff
@@ -28,9 +28,10 @@
+
 import txzmq
 
-from leap.common.events.zmq_components import TxZmqClientComponent
```
If there was not reason for the ordering I guess this change is neither needed.

kalikaneko replied:
fixed

meskio commented on src/leap/common/events/zmq_components.py:
```diff
@@ -14,60 +14,61 @@
 from leap.common.zmq_utils import maybe_create_and_get_certificates
 from leap.common.zmq_utils import PUBLIC_KEYS_PREFIX
 
+from leap.common.events.auth import TxAuthenticator, TxAuthenticationRequest
```
This is going to import zmq.auth, which in file is imported in a try/except (I guess to work on envs without libsodium). Is not going to throw an exception in some platforms this import? If not, why do we keep zmq.auth in a try/except?

kalikaneko replied:
you're right, this should be inside the try/except block

kalikaneko replied:
fixed

meskio commented on src/leap/common/events/txclient.py:
```diff
@@ -122,7 +124,6 @@ def _run_callback(self, callback, event, content):
         callback(event, *content)
 
     def shutdown(self):
-        TxZmqClientComponent.shutdown(self)
```
You don't need to do the shutdown anymore?

kalikaneko replied:
See https://github.com/smira/txZMQ/blob/master/txzmq/factory.py#L44
Since I'm calling the registerForShutdown on the factory, and the factory keeps track of the connections, I see no need of doing this ourselves.

meskio commented on src/leap/common/events/zmq_components.py:
```diff
@@ -118,64 +120,40 @@ def _zmq_bind(self, connClass, address):
-        # XXX do not hardcode this here.
-        authenticator.allow('127.0.0.1')
-        # tell authenticator to use the certificate in a directory
+        # tell authenticator to use the certificate in a directory 
```
I think this commit is adding a trailing space here.

kalikaneko replied:
fixed

kalikaneko commented:
I have moved the test_events to the events.tests module, and modified it so that it can run properly with the latest modifications. all tests passing. 
also added a small test_auth for the curve authentication.
